packages/date-picker/src/vaadin-date-picker-overlay-content.js

```diff
@@ -413,11 +413,13 @@ class DatePickerOverlayContent extends ThemableMixin(DirMixin(PolymerElement)) {
    */
   revealDate(date) {
     if (date) {
-      var diff = this._differenceInMonths(date, this._originDate);
-      var scrolledAboveViewport = this.$.monthScroller.position > diff;
+      const diff = this._differenceInMonths(date, this._originDate);
+      const scrolledAboveViewport = this.$.monthScroller.position > diff;
 
-      var visibleItems = this.$.monthScroller.clientHeight / this.$.monthScroller.itemHeight;
-      var scrolledBelowViewport = this.$.monthScroller.position + visibleItems - 1 < diff;
+      const visibleItems = Math.floor(
```

Comment:
There are several alternatives:

## 👍 With `Math.floor`:

```js
const visibleItems = Math.floor(
  (monthScroller.clientHeight - monthScroller.bufferOffset) / monthScroller.itemHeight
);
```

When scrolling down, this ensures the first visible month has the buffer offset relative to the calendar's top:

## 👍 With an additional buffer offset:

```js
const visibleItems = (monthScroller.clientHeight - monthScroller.bufferOffset * 2) / monthScroller.itemHeight;
```

When scrolling down, this ensures the last visible month has the buffer offset relative to the calendar's bottom:

Uploading Screen Recording 2021-12-24 at 14.07.23.mov…

## 👎 Without `Math.floor` and an additional offset: 

When scrolling down, I noticed that, with this configuration, the last visible month occasionally slightly goes into transparency at the bottom. I don't think it is fine.

```js
const visibleItems = (monthScroller.clientHeight - monthScroller.bufferOffset) / monthScroller.itemHeight;
```

Reply:
I would personally prefer (2). This option seems optimal to me in terms of how much the calendar area is scrolled: the less is scrolled, the better it is for the user.

cc @tomivirkki @web-padawan @jouni

Reply:
Option 2 looks the best from these, I agree.